doc: Bump to Ubuntu Bionic 18.04 in build-windows.md

ken2812221 opened the pull request:
Windows starts to provide Ubuntu 18.04 WSL, it can be downloaded from Microsoft Store.

ken2812221 committed:
doc: Bump to Ubuntu Bionic 18.04 in build-windows.md

Windows starts to provide Ubuntu 18.04 WSL, it can be downloaded from Microsoft Store.

promag commented:
Concept ACK.

There are more references to 17.10 that could be bumped?

fanquake commented:
Concept ACK
This will need a few more updates though. For example Ubuntu and Ubuntu 18.04 are two different items in the Windows Store, so we should link to 18.04 from here.

MarcoFalke commented:
I had the impression that we only support windows cross builds for bionic, due to qt 5.9. So might as well remove all previous versions/workarounds?

ken2812221 commented:
According to @MarcoFalke's comment, removed all other distros except for Ubuntu 18.04. Also changed the Ubuntu 18.04 WSL link according to @fanquake's comment.

fanquake commented:
Thanks @ken2812221 . Instructions are looking simpler now. Was going to run through a build but VirtualBox has crashed twice while building depends using 18.04 WSL.. Will ACK once I've got a build finished.

ken2812221 commented:
Update: Search "Ubuntu 18.04"

laanwj commented:
Concept ACK. I like this as it gets rid of a lot of confusing backwards compatibility cruft.

laanwj commented on doc/build-windows.md:
```diff
@@ -38,10 +38,10 @@ To install WSL on Windows 10 with Fall Creators Update installed (version >= 162
   * Click Install
 3. Complete Installation
-  * Open a cmd prompt and type "Ubuntu"
+  * Open a cmd prompt and type "Ubuntu1804"
```
Will this stay this way? Or is this temporarily, until `1804` becomes the default

ken2812221 replied:
I'm not sure, it's a different app in Microsoft Store now.

fanquake replied:
There are multiple different Ubuntu "Apps" available in the Windows store, so I think this should be ok for now.

laanwj commented on doc/build-windows.md:
```diff
@@ -5,7 +5,7 @@ Below are some notes on how to build Bitcoin Core for Windows.
 The options known to work for building Bitcoin Core on Windows are:
 
-* On Linux using the [Mingw-w64](https://mingw-w64.org/doku.php) cross compiler tool chain. Ubuntu Trusty 14.04 is recommended
+* On Linux using the [Mingw-w64](https://mingw-w64.org/doku.php) cross compiler tool chain. Ubuntu Bionic 18.04 is recommended
```
Please add that these steps only apply for `18.04`. Only saying that it is recommended is not enough, as it means you'd still have to warn people against using `16.04`.

ken2812221 replied:
Change `recommended` to `required`

fanquake commented:
TestedACK https://github.com/bitcoin/bitcoin/pull/13246/commits/9d4f9421a4898f75a4c49aaefaba2868de071d79

This is a good cleanup, less workarounds.

I've done a 64 bit depends build using WSL on a clean Windows 10 VM @ https://github.com/bitcoin/bitcoin/commit/6916024768ec57a00f54224640ab4e4871d2a30a
![wsl](https://user-images.githubusercontent.com/863730/40425978-75011ffa-5ecc-11e8-89f2-b1287ce0fd00.png)

MarcoFalke commented:
utACK 9d4f9421a4898f75a4c49aaefaba2868de071d79
